qiskit/extensions/standard/cswap.py

```diff
@@ -15,19 +15,19 @@
 """
 Fredkin gate. Controlled-SWAP.
 """
-from qiskit.circuit import Gate
+from qiskit.circuit import ControlledGate
 from qiskit.circuit import QuantumCircuit
 from qiskit.circuit import QuantumRegister
 from qiskit.extensions.standard.cx import CnotGate
 from qiskit.extensions.standard.ccx import ToffoliGate
 
 
-class FredkinGate(Gate):
+class FredkinGate(ControlledGate):
     """Fredkin gate."""
 
     def __init__(self):
         """Create new Fredkin gate."""
-        super().__init__("cswap", 3, [])
+        super().__init__("cswap", 3, [], num_ctrl_qubits=1)
 
```

Comment:
add `self.base_gate` here for the cswap gate?

Reply:
yes